torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -2741,6 +2741,59 @@ class TestPickler:
             torch.distributed.rpc.api._default_pickler is _internal_rpc_pickler
         )
 
+    @dist_init
+    def test_wait_all(self):
+        with rpc.wait_all():
+            self.assertTrue(_thread_local_var.future_list == [])
+            dst = worker_name((self.rank + 1) % self.world_size)
+            fut = rpc.rpc_async(dst, torch.add, (torch.ones(2, 2), 1))
+            self.assertFalse(fut.done())
```

Comment:
this can potentially be flaky, in case the `fut` is done before the check? Maybe remove this?